From c394d70b6d62e2d414f01102134c603779e5f37a Mon Sep 17 00:00:00 2001
From: Michael D Labriola <mdlabriola@yahoo.com>
Date: Thu, 14 Apr 2011 19:44:05 +0000
Subject: [PATCH] configure.ac: remove ENABLE_CRYPTO checks

This commit removes the entire block of code that checks for libgcrypt and
gnutls.  We don't want the ntfsprogs crypto stuff to be built for
bootstrap, and we don't want to have to add gcrypt to bootstrap just so the
AM_PATH_LIBGCRPT macro will be defined...

This patch should not be applied to our main source tree.  Technically, it
probably could be, because I don't think we're ever gonna want the
ntfsprogs crypto stuff to be enabled... but why branch off of mainline when
we really don't need to.  Just make sure that libgcrypt gets installed
prior to bootstrapping the ntfs-3g source tree for the final build.
---
 configure.ac |   27 ---------------------------
 1 files changed, 0 insertions(+), 27 deletions(-)

diff --git a/configure.ac b/configure.ac
index 8bc9bc2..c595b2e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -276,33 +276,6 @@ fi
 
 # Autodetect whether we can build crypto stuff or not.
 compile_crypto=false
-if test "$enable_crypto" != "no"; then
-	have_libgcrypt=false
-	AM_PATH_LIBGCRYPT(1.2.2, [ have_libgcrypt=true ],
-	[
-		if test "$enable_crypto" = "yes"; then
-			AC_MSG_ERROR([ntfsprogs crypto code requires the gcrypt library.])
-		else
-			AC_MSG_WARN([ntfsprogs crypto code requires the gcrypt library.])
-		fi
-	])
-	have_libgnutls=false
-	PKG_CHECK_MODULES(GNUTLS, gnutls >= 1.4.4, [ have_libgnutls=true ],
-		if test "$enable_crypto" = "yes"; then
-			AC_MSG_ERROR([ntfsprogs crypto code requires the gnutls library.])
-		else
-			AC_MSG_WARN([ntfsprogs crypto code requires the gnutls library.])
-		fi
-	)
-	if test "$have_libgcrypt" = "true"; then
-		if test "$have_libgnutls" = "true"; then
-			compile_crypto=true
-			AC_DEFINE([ENABLE_CRYPTO], 1,
-			[Define this to 1 if you want to enable support of
-			 encrypted files in libntfs and utilities.])
-		fi
-	fi
-fi
 AM_CONDITIONAL(ENABLE_CRYPTO, $compile_crypto)
 
 # add --with-extra-includes and --with-extra-libs switch to ./configure
-- 
1.6.0.3

